### PR TITLE
cdk publish: add required token to commit to master and --no-cache for getting just-published version from pypi

### PR DESCRIPTION
**Pull request by erohmensing**
Some fixes to the workflow: 
* we need a token when checking out to be able to commit back to master
* Without `--no-cache`, poetry's pypi cache immediately decides it can't find the new version that we just published in the previous step. `--no-cache` forces it to pull the versions again. It's possible that by the time it does this, pypi won't have updated it yet in their apis - if so we'll need to add a timeout too. But we'll try this first.

### CODE REVIEW

**Comment by erohmensing** [comment]
* **#36864** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36864?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @erohmensing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by octavia-approvington** [comment]
After a careful ML study, <br/> we think this looks okay.  <br/> imagine code being okay

**Review by octavia-approvington — APPROVED** [review]

File: .github/workflows/publish-cdk-command-manually.yml [hunk 1]
```diff
@@ -97,7 +97,7 @@ jobs:
         with:
           repository: ${{ github.event.inputs.repo }}
           ref: ${{ github.event.inputs.gitref }}
-          token: ${{ secrets.GH_PAT_MAINTENANCE_OSS }}
+          token: ${{ secrets.GH_PAT_MAINTENANCE_OSS }} # This token is what allows us to commit directly to master
       - name: "Publish Python CDK: bump Poetry package version"
         id: bumpversion
         run: |
```
Comment (natikgadzhi) [inline .github/workflows/publish-cdk-command-manually.yml:103]: Oh I love this, now I know where we have the logic that pushes the version into the Changelog.

**Review by natikgadzhi — COMMENTED** [review]
Thank you for the cleanup!

**Pull request merged**